src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,17 @@
+package duke;
+
+import duke.ui.MainWindow;
+
+public class DukeException extends Exception {
+
+    public static MainWindow mw;
+
+    /**
+     * Creates a Duke Exception object.
+     *
+     * @param message Error message to print.
+     */
+    public DukeException(String message) {
+        mw.setDukeMessage("\u2639 OOPS!!! " + message);
```

Comment:
I think having the `DukeException` class print messages to a static member in the constructor may potentially be confusing for some developers who are more used to the convention of using try/catch blocks to handle exceptions.